plugins/cairo-lang-macro/src/lib.rs

```diff
@@ -24,6 +25,28 @@ pub unsafe extern "C" fn free_result(result: StableProcMacroResult) {
     ProcMacroResult::from_owned_stable(result);
 }
 
+#[distributed_slice]
+pub static AUX_DATA_CALLBACKS: [fn(Vec<AuxData>)];
+
+#[no_mangle]
+#[doc(hidden)]
+pub unsafe extern "C" fn aux_data_callback(
+    stable_aux_data: StableSlice<StableAuxData>,
+) -> StableSlice<StableAuxData> {
+    if !AUX_DATA_CALLBACKS.is_empty() {
+        // Callback has been defined, applying the aux data collection.
+        let fun = AUX_DATA_CALLBACKS[0];
```

Comment:
How do we want to deal with devs making multiple instances of aux_data_callback?